From 8c135d275d73575140410b7f1a2d7e28cc790a83 Mon Sep 17 00:00:00 2001
From: Vishalcj17 <vishalcj@aospa.co>
Date: Sun, 4 Oct 2026 11:08:04 +0700
Subject: [PATCH] ChooseLockPassword: Decrease minimum pin length for auto
 confirm

This default enables auto confirm on a pin length of 4 and
correctly displays it in setup

Change-Id: I482be235261de1e636c89e33722df4f4e412b07c
(cherry picked from commit 1ed39d4f63f5cbf48f1b0996c278f7d9fc3ef783)

Adapt to Lineage:
- Add lock_screen_auto_pin_confirm_summary_custom string (4 digits),
  not present in this base

Change-Id: I2b8748b18792ab66bde1ce3fdb4f6b651fb529ab
---
 res/values/strings.xml                                    | 2 ++
 res/xml/screen_lock_settings.xml                          | 2 +-
 src/com/android/settings/password/ChooseLockPassword.java | 4 ++--
 3 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/res/values/strings.xml b/res/values/strings.xml
index ac692e4..ad095c0 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -1718,6 +1718,8 @@
     <string name="lock_screen_auto_pin_confirm_title">Auto-confirm unlock</string>
     <!-- Summary of the lock screen auto pin confirm setting. [CHAR LIMIT=NONE] -->
     <string name="lock_screen_auto_pin_confirm_summary">Unlock automatically if you input a correct PIN of 6 digits or more. This is slightly less secure than tapping Enter to confirm.</string>
+    <!-- Summary of the lock screen auto pin confirm setting with reduced minimum length. [CHAR LIMIT=NONE] -->
+    <string name="lock_screen_auto_pin_confirm_summary_custom">Unlock automatically if you input a correct PIN of 4 digits or more. This is slightly less secure than tapping Enter to confirm.</string>
     <!-- Message shown to check auto pin confirmation feature when the user is updating the PIN. [CHAR LIMIT=NONE] -->
     <string name="auto_pin_confirm_user_message">Auto-confirm correct PIN</string>
     <!-- Message shown to explain the security concern if a user opts-in to the auto-pin feature. [CHAR LIMIT=NONE] -->
diff --git a/res/xml/screen_lock_settings.xml b/res/xml/screen_lock_settings.xml
index 19061d9..cfd4c1e 100644
--- a/res/xml/screen_lock_settings.xml
+++ b/res/xml/screen_lock_settings.xml
@@ -31,7 +31,7 @@
     <SwitchPreferenceCompat
         android:key="auto_pin_confirm"
         android:title="@string/lock_screen_auto_pin_confirm_title"
-        android:summary="@string/lock_screen_auto_pin_confirm_summary" />
+        android:summary="@string/lock_screen_auto_pin_confirm_summary_custom" />
 
     <SwitchPreferenceCompat
         android:key="enhancedPinPrivacy"
diff --git a/src/com/android/settings/password/ChooseLockPassword.java b/src/com/android/settings/password/ChooseLockPassword.java
index 5496b14..6f4a594 100644
--- a/src/com/android/settings/password/ChooseLockPassword.java
+++ b/src/com/android/settings/password/ChooseLockPassword.java
@@ -252,7 +252,7 @@ public class ChooseLockPassword extends SettingsActivity {
         private static final String KEY_IS_AUTO_CONFIRM_CHECK_MANUALLY_CHANGED =
                 "auto_confirm_option_set_manually";
 
-        private static final int MIN_AUTO_PIN_REQUIREMENT_LENGTH = 6;
+        private static final int MIN_AUTO_PIN_REQUIREMENT_LENGTH = 4;
 
         private LockscreenCredential mCurrentCredential;
         private LockscreenCredential mChosenPassword;
@@ -1125,7 +1125,7 @@ public class ChooseLockPassword extends SettingsActivity {
                 mAutoPinConfirmOption.setVisibility(View.VISIBLE);
                 mAutoConfirmSecurityMessage.setVisibility(View.VISIBLE);
                 if (!mIsAutoPinConfirmOptionSetManually) {
-                    mAutoPinConfirmOption.setChecked(length == MIN_AUTO_PIN_REQUIREMENT_LENGTH);
+                    mAutoPinConfirmOption.setChecked(length >= MIN_AUTO_PIN_REQUIREMENT_LENGTH);
                 }
             } else {
                 mAutoPinConfirmOption.setVisibility(View.GONE);
